### PR TITLE
Move some model classes to Protobuf classes [4/?]

### PR DESCRIPTION
**Pull request by prm-james-hill**

### CODE REVIEW

File: src/main/java/ai/promoted/delivery/client/PromotedDeliveryClient.java [hunk 1]
```diff
@@ -283,7 +283,7 @@ private boolean shouldApplyTreatment(CohortMembership cohortMembership) {
       return true;
     }
 
-    return cohortMembership.getArm() == null || cohortMembership.getArm() != CohortArm.CONTROL;
+    return cohortMembership.getArm() != CohortArm.CONTROL;
```
Comment (prm-james-hill) [inline src/main/java/ai/promoted/delivery/client/PromotedDeliveryClient.java:286]: Tempted to add an `&& arm != UNKNOWN` here

Comment (prm-james-hill) [reply src/main/java/ai/promoted/delivery/client/PromotedDeliveryClient.java:286]: Will re-consider when I do another pass at a higher level

File: src/test/java/ai/promoted/delivery/client/PromotedDeliveryClientTest.java [hunk 2]
```diff
@@ -253,74 +253,74 @@ void testHasControlCohortMembershipCallsSDKDeliveryAndLogs() throws Exception {
...
   void testPassThroughSdkInsertionIds() throws Exception {
     PromotedDeliveryClient client = createDefaultClient();
 
-    Request req = new Request().insertion(TestUtils.createTestRequestInsertions(10));
-    List<Insertion> insertions = req.getInsertion();
-    for (int i = 0; i < insertions.size(); i++) {
-      insertions.set(i, insertions.get(i).toBuilder().setInsertionId("ins" + i).build());
+    Request.Builder reqBuilder = Request.newBuilder().addAllInsertion(TestUtils.createTestRequestInsertions(10));
+    List<Insertion.Builder> insBuilders = reqBuilder.getInsertionBuilderList();
+    for (int i = 0; i < insBuilders.size(); i++) {
```
Comment (prm-dan) [inline src/test/java/ai/promoted/delivery/client/PromotedDeliveryClientTest.java:310]: Nit - I'd guess you don't need the extra list.  You can probably use getInsertionBuilder(index).

**Review by prm-dan — APPROVED** [review]

File: src/main/java/ai/promoted/delivery/client/PromotedDeliveryClient.java [hunk 3]
```diff
@@ -364,27 +364,27 @@ private LogRequest createLogRequest(DeliveryRequest deliveryRequest, Response re
    *
    * @param request the request to populate
    */
-  private void fillInRequestFields(Request request) {
-    if (request.getClientInfo() == null) {
-      request.setClientInfo(ClientInfo.newBuilder().build());
+  private void fillInRequestFields(Request.Builder requestBuilder) {
+    if (!requestBuilder.hasClientInfo()) {
+      requestBuilder.setClientInfo(ClientInfo.newBuilder().build());
```
[the remark this answers is not in the document]
Comment (prm-james-hill) [reply src/main/java/ai/promoted/delivery/client/PromotedDeliveryClient.java:369]: Ah, realized but missed this instance. Fixed

File: src/test/java/ai/promoted/delivery/client/ApiDeliveryTest.java [hunk 4]
```diff
@@ -2,21 +2,18 @@
...
-    response.setRequestId("reqid");
-    ApiDelivery.validate(response);
+    ApiDelivery.validate(Response.newBuilder().setRequestId("reqid").build());
   }
 
   @Test
   void validate_missingRequestId() {
-    Response response = new Response();
-    assertThrows(DeliveryException.class, () -> ApiDelivery.validate(response));
+    assertThrows(DeliveryException.class, () -> ApiDelivery.validate(Response.newBuilder().build()));
```
[the remark this answers is not in the document]
Comment (prm-james-hill) [reply src/test/java/ai/promoted/delivery/client/ApiDeliveryTest.java:16]: Fixed. I actually looked for this but chatgpt confidently told me newBuilder().build() was idiomatic

File: src/test/java/ai/promoted/delivery/client/PromotedDeliveryClientTest.java (hunk 2 above)
Comment (prm-james-hill) [reply src/test/java/ai/promoted/delivery/client/PromotedDeliveryClientTest.java]: Done

**Pull request merged**